### PR TITLE
Refresh Python versions

### PR DESCRIPTION
**Pull request by ericzolf**
## Changes done and why

CHG: remove support for Python below 3.8 and add proper Python 3.11 support, also under Windows
CHG: remove support for 32 bits Linux by lack of working manylinux image (you can probably still compile it yourself locally)

## Self-Checklist

- [x] changes to the code have been reflected in the documentation
- [x] changes to the code have been covered by new/modified tests
- [x] commit contains a description of changes relevant to users prefixed by DOC:, FIX:, NEW: and/or CHG:

### CODE REVIEW

**Comment by ericzolf** [comment]
@ikus060 this is the kind of changes required to make 3.11 work, it is for me too extensive for a patch version, hope this is OK for you.

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -26,14 +26,14 @@ jobs:
...
-        #python-version: ["3.11","3.10","3.9","3.8","3.7"]
+        #python-version: ["3.11","3.10","3.9","3.8"]
         #--- Build and deploy Linux wheels using manylinux containers ---
         # - build manylinux2010 x64 and i686
         # - build manylinux2014 x64
         # - build manylinux_2_28 x64
         # avoiding manylinux2014_i686 because does not provide librsync-devel
         # manylinux_2_24 is already deprecated
-        many-linux: [2010_x86_64, 2010_i686, 2014_x86_64, _2_28_x86_64]
+        many-linux: [2014_x86_64, _2_28_x86_64]
```
Comment (ikus060) [inline .github/workflows/deploy.yml:36]: I'm not fluent with ManyLinux. Correct me if I'm wrong. I tought the new convention is to use _2_28_x86_64 where _2_28 is compatible with glibc >=2.28. Is it required to keep 2014_x86_64 ?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -47,7 +47,7 @@ jobs:
           if [[ ${{ matrix.many-linux }} != *64 ]]; then PRE_CMD=linux32; fi
           #py=$(echo ${{ matrix.python-version }} | tr -d .)
           plat=manylinux${{ matrix.many-linux }}
-          $PRE_CMD /ws/tools/build_wheels.sh /ws ${plat} /opt/python/cp3{11,10,9,8,7}*/bin
+          $PRE_CMD /ws/tools/build_wheels.sh /ws ${plat} /opt/python/cp3{11,10,9,8}*/bin
```
Comment (ikus060) [inline .github/workflows/deploy.yml:50]: We should probably keep compatibility with python 3.7.

Debian Buster is supported until June 30th, 2024 and is release with python 3.7
If it's not too hard. https://wiki.debian.org/LTS

File: .github/workflows/test_windows.yml [hunk 3]
```diff
@@ -17,7 +17,7 @@ defaults:
     shell: bash
 
 env:
-  WIN_PYTHON_VERSION: 3.10.7
+  WIN_PYTHON_VERSION: 3.11.2
```
Comment (ikus060) [inline .github/workflows/test_windows.yml:20]: I guess there is not such thing as manywindow !? It's a shame that only one binary version is distributed for Windows.

File: requirements.txt [hunk 4]
```diff
@@ -28,7 +27,7 @@ pywin32 ; platform_system=="Windows"
 twine
 wheel
 #py2exe; platform_system=="Windows"
-pyinstaller==5.4; platform_system=="Windows"
+pyinstaller; platform_system=="Windows"
```
Comment (ikus060) [inline requirements.txt:30]: I strongly suggest to fix the version here. I had a couple of problem from time to time with pyinstaller. They keep changing the bootloader and the behavior is not the same. e.g.: the latest version was not resolving call to symbolic link. So creating a symlink of rdiff-backup in /usr/bin was not working.

File: .github/workflows/test_windows.yml (hunk 3 above)
Comment (ericzolf) [reply .github/workflows/test_windows.yml:20]: Well, the difference between Windows and Linux is that Python comes with Linux, so you're dependent on the version delivered (more or less). On Windows, you anyway need to install Python and can decide which version (and we deliver anyway a compiled binary).

File: requirements.txt (hunk 4 above)
Comment (ericzolf) [reply requirements.txt:30]: Yeah, on the other hand, it led to having a shit old version of it, being a few versions behind, so not optimal.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (ericzolf) [reply .github/workflows/deploy.yml:50]: No, sorry, the rule is to have 4 supported versions of Python, that takes enough of my time. Debian people will have to stay on 2.2, if they are clever enough to install their own version of rdiff-backup, they will probably be clever enough to install their own version of Python, else they go with what's delivered with their platform.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (ericzolf) [reply .github/workflows/deploy.yml:36]: My understanding is, yes, for slightly older platforms: https://github.com/pypa/manylinux (e.g. CentOS 8 with Python3.8).

File: requirements.txt (hunk 4 above)
Comment (ericzolf) [reply requirements.txt:30]: After thinking about it, I prefer to cope with small incompatibilities on a regular basis, than a bunch of incompatibilities all at once, not knowing which version of PyInstaller introduced it, so I stick to my decision to remove the fixed version.

**Comment by ericzolf** [comment]
All these comments answered, I'll have to throw away this PR anyway, so that I can fix #850 which is a bit too blunt to my taste to keep it in a stable version *sigh*

**Pull request closed without merging**